### ISSUE TITLE
errors/local.xml and error page templates are publicly accessible

### ISSUE THREAD

**Issue by schmengler** [magento/magento2#20209]
### Preconditions (*)

All current Magento 2 versions:
1. 2.3.0
2. 2.2.7
3. 2.1.16

Using default apache or nginx configuration with `pub` as doc root

### Steps to reproduce (*)

For PHTML files:
1. Point browser to /errors/default/page.phtml 

For local.xml:
1. Copy pub/errors/local.xml.sample to pub/errors/local.xml
2. Point browser to /errors/local.xml

### Expected result (*)

1. A "not found" or "forbidden" response

### Actual result (*)

1. The source files are served

**Comment by magento-engcom-team** [comment]
Hi @schmengler. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@schmengler do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by schmengler** [comment]
@magento-engcom-team give me 2.3.0 instance

**Comment by magento-engcom-team** [comment]
Hi @schmengler. Thank you for your request. I'm working on Magento 2.3.0 instance for you

**Comment by magento-engcom-team** [comment]
Hi @schmengler, here is your [Magento instance](https://i-20209-2-3-0.instances.magento-community.engineering).
Admin access: [https://i-20209-2-3-0.instances.magento-community.engineering/admin](https://i-20209-2-3-0.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ghost** [comment]
hi @schmengler did you configure docroot to pub ?

**Comment by schmengler** [comment]
@engcom-backlog-nazar yes, I should have mentioned it, sorry.

In the instance above it seems not to be the case, but I can access for example: https://i-20209-2-3-0.instances.magento-community.engineering/pub/errors/local.xml.sample

local.xml does not exist on vanilla instance and PHTML files seem to be parsed as PHP, so for example https://i-20209-2-3-0.instances.magento-community.engineering/pub/errors/page.phtml results in
```
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml" >
<head>
    <title>
```
it terminates with an error at `<?= $this->pageTitle ?>`

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97575`, `MAGETWO-97576` were created

**Comment by magento-engcom-team** [comment]
Hi @schmengler. Thank you for your report. 
The issue has been fixed in magento/magento2#20212 by @schmengler in 2.3-develop branch
Related commit(s):
 - [a0566abffe872cca0a7e06eb2b4d08093f15fbda](https://github.com/magento/magento2/commit/a0566abffe872cca0a7e06eb2b4d08093f15fbda)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @schmengler. Thank you for your report. 
The issue has been fixed in magento/magento2#21946 by @amol2jcommerce in 2.2-develop branch
Related commit(s):
 - [2d291a73a65a6009ce99edf32569e484e6fc3a19](https://github.com/magento/magento2/commit/2d291a73a65a6009ce99edf32569e484e6fc3a19)

The fix will be available with the upcoming 2.2.9 release.
